packages/dev/core/src/Physics/v2/physicsConstraint.ts

```diff
@@ -0,0 +1,270 @@
+     *
+     */
+    public getAxisMaxLimit(axis: ConstraintAxis): number {
+        return this._physicsPlugin ? this._physicsPlugin.getAxisMaxLimit(this, axis) : 0;
+    }
+
+    /**
+     *
+     */
+    public setAxisMotorType(axis: ConstraintAxis, motorType: ConstraintMotorType): void {
+        this._physicsPlugin?.setAxisMotorType(this, axis, motorType);
+    }
+
+    /**
+     *
+     */
+    public getAxisMotorType(axis: ConstraintAxis): ConstraintMotorType {
+        return this._physicsPlugin ? this._physicsPlugin.getAxisMotorType(this, axis) : ConstraintMotorType.NONE;
+    }
+
+    /**
+     *
+     */
+    public setAxisMotorTarget(axis: ConstraintAxis, target: number): void {
+        this._physicsPlugin?.setAxisMotorTarget(this, axis, target);
+    }
+
+    /**
+     *
+     */
+    public getAxisMotorTarget(axis: ConstraintAxis): number {
+        return this._physicsPlugin ? this._physicsPlugin.getAxisMotorTarget(this, axis) : 0;
+    }
+
+    /**
+     *
+     */
+    public setAxisMotorMaxForce(axis: ConstraintAxis, maxForce: number): void {
+        this._physicsPlugin?.setAxisMotorMaxForce(this, axis, maxForce);
+    }
+
+    /**
+     *
+     */
+    public getAxisMotorMaxForce(axis: ConstraintAxis): number {
+        return this._physicsPlugin ? this._physicsPlugin.getAxisMotorMaxForce(this, axis) : 0;
+    }
+
+    /**
+     *
+     */
+    public dispose(): void {
+        this._physicsPlugin?.disposeConstraint(this);
+    }
+}
+
+/**
+ *
+ */
+export class PhysicsConstraintBallAndSocket extends PhysicsConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, scene: Scene) {
+        super(ConstraintType.BALL_AND_SOCKET, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
+    }
+}
+
+/**
+ *
+ */
+export class PhysicsConstraintDistance extends PhysicsConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, scene: Scene) {
+        super(ConstraintType.DISTANCE, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
+    }
+}
+
+/**
+ *
+ */
+export class PhysicsConstraintHinge extends PhysicsConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, scene: Scene) {
+        super(ConstraintType.HINGE, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
+    }
+}
+
+/**
+ *
+ */
+export class PhysicsConstraintSlider extends PhysicsConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, scene: Scene) {
+        super(ConstraintType.SLIDER, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
+    }
+}
+
+/**
+ *
+ */
+export class PhysicsConstraintLock extends PhysicsConstraint {
+    constructor(pivotA: Vector3, pivotB: Vector3, axisA: Vector3, axisB: Vector3, scene: Scene) {
+        super(ConstraintType.LOCK, { pivotA: pivotA, pivotB: pivotB, axisA: axisA, axisB: axisB }, scene);
+    }
+}

```

Comment:
Why are we passing an `Options` object to the parent class, but individual parameters here? can't you reuse the Options object?

Another question - is this really needed? Can't we put the constraint type inside the options and take it out of the constructor?

Reply:
This is a placeholder: idk yet in which proportion constraints will have different parameters. If they all end up having same parameters (2 points + 2 axis) then I will use a unique option parameter object.

Reply:
You can add the `@beta` or `@experimental` tag so that we won't have any issue with back-compat when they change

Reply:
Or `@internal` for now?

Reply:
beta is too advanced for the current state. @internal seems to fit better for now

Reply:
done.